src/pixelator/graph/backends/implementations.py

```diff
@@ -428,9 +517,118 @@ def __init__(
         """
         self._raw = raw
 
+    @staticmethod
+    def _build_plain_graph_from_edgelist(
+        df: pl.LazyFrame,
+        create_using: Union[nx.Graph, nx.MultiGraph],
+    ) -> Union[nx.Graph, nx.MultiGraph]:
+        g = nx.empty_graph(0, create_using)
+        g.add_edges_from(
+            (row[0], row[1])
+            for row in df.collect(streaming=True).iter_rows(
+                named=False, buffer_size=1000
+            )
+        )
+        return g
+
+    @staticmethod
+    def _build_graph_with_node_counts_from_edgelist(
+        df: pl.LazyFrame,
+        create_using: bool,
+    ) -> Union[nx.Graph, nx.MultiGraph]:
+        unique_markers = set(df.unique("marker").collect()["marker"].to_list())
+        initial_marker_dict = {marker: 0 for marker in unique_markers}
+
+        g: nx.Graph = nx.empty_graph(0, create_using)
+
+        for row in (
+            df.select(["upia", "upib", "marker"])
+            .collect(streaming=True)
+            .iter_rows(named=False, buffer_size=1000)
+        ):
+            # We are duplicating code here, since it gives
```

Comment:
What is the part of the code and duplication that is giving this performance boost? Is not entirely clear to me.

Reply:
Previously there was a function that was used here. @fbdtemme suggested inlineing it: https://github.com/PixelgenTechnologies/pixelator/pull/32#discussion_r1372827702 so now there is a bit more code duplication there.